sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java

```diff
@@ -0,0 +1,220 @@
+   * @return The SentryAppender.
+   */
+  @PluginFactory
+  public static SentryAppender createAppender(
+      @PluginAttribute("name") final String name,
+      @PluginAttribute("minimumBreadcrumbLevel") final Level minimumBreadcrumbLevel,
+      @PluginAttribute("minimumEventLevel") final Level minimumEventLevel,
+      @PluginAttribute("dsn") final String dsn,
+      @PluginElement("filter") final Filter filter) {
+
+    if (name == null) {
+      LOGGER.error("No name provided for SentryAppender");
+      return null;
+    }
+    return new SentryAppender(name, filter, dsn, minimumBreadcrumbLevel, minimumEventLevel, null);
+  }
+
+  @Override
+  public void start() {
+    if (dsn != null) {
+      Sentry.init(
+          options -> {
+            options.setDsn(dsn);
+            options.setSentryClientName(BuildConfig.SENTRY_LOG4J2_SDK_NAME);
+            options.setSdkVersion(createSdkVersion(options));
+            Optional.ofNullable(transport).ifPresent(options::setTransport);
+          });
+    }
+    super.start();
+  }
+
+  @Override
+  public void append(LogEvent eventObject) {
+    if (eventObject.getLevel().isMoreSpecificThan(minimumEventLevel)) {
+      Sentry.captureEvent(createEvent(eventObject));
+    }
+    if (eventObject.getLevel().isMoreSpecificThan(minimumBreadcrumbLevel)) {
+      Sentry.addBreadcrumb(createBreadcrumb(eventObject));
+    }
+  }
+
+  /**
+   * Creates {@link SentryEvent} from Log4j2 {@link LogEvent}.
+   *
+   * @param loggingEvent the log4j2 event
+   * @return the sentry event
+   */
+  @SuppressWarnings("JdkObsolete")
+  final @NotNull SentryEvent createEvent(@NotNull LogEvent loggingEvent) {
+    final SentryEvent event =
+        new SentryEvent(DateUtils.getDateTime(new Date(loggingEvent.getTimeMillis())));
+    final Message message = new Message();
+    message.setMessage(loggingEvent.getMessage().getFormat());
+    message.setFormatted(loggingEvent.getMessage().getFormattedMessage());
+    message.setParams(toParams(loggingEvent.getMessage().getParameters()));
+    event.setMessage(message);
+    event.setLogger(loggingEvent.getLoggerName());
+    event.setLevel(formatLevel(loggingEvent.getLevel()));
+
+    final ThrowableProxy throwableInformation = loggingEvent.getThrownProxy();
+    if (throwableInformation != null) {
+      event.setThrowable(throwableInformation.getThrowable());
+    }
+
+    if (loggingEvent.getThreadName() != null) {
+      event.setExtra("thread_name", loggingEvent.getThreadName());
+    }
+
+    final Map<String, String> mdcProperties =
+        CollectionUtils.shallowCopy(loggingEvent.getContextData().toMap());
+    if (!mdcProperties.isEmpty()) {
+      event.getContexts().put("MDC", mdcProperties);
+    }
+
+    return event;
+  }
+
+  private @NotNull List<String> toParams(@Nullable Object[] arguments) {
+    if (arguments != null) {
+      return Arrays.stream(arguments)
+          .filter(Objects::nonNull)
+          .map(Object::toString)
```

Comment:
Is it toString what the formatter call on the args? If not, this could end up being a problem.
Since we already give both formatted and template, it's probably ok not to add the arga at all.

Sentry would group on the template and display the formatted already

Reply:
Adding extra args result in following entries in Sentry UI
![image](https://user-images.githubusercontent.com/1357927/91977377-e1221680-ed22-11ea-8af8-ddfa110a75d0.png)

If we do not need it, we can get rid of passing parameters.

Reply:
It's nice to have, sure. I just ask because if the Formatting code calls some `format()` method or else, instead of `toString()` the value we return in the `args` of the Sentry event won't match the actual formatted string log message

Reply:
Potentially they won't match in some special cases. I don't think it makes sense to decode https://github.com/apache/logging-log4j2/blob/master/log4j-api/src/main/java/org/apache/logging/log4j/message/ParameterFormatter.java to make sure.

Up to you - we can remove it.

Reply:
Let's keep it then, I assume worst case is returning the default Object's toString which is useless. Hopefully no massive blob of data